scripts/automation/lecterntrade.arucas

```diff
@@ -0,0 +1,488 @@
+    SyL = getIterator(sy-reach, sy+reach, 1);
+    SzL = getIterator(sz-reach, sz+reach, 1);
+    //first prioritize looking direction
+    checkPos = getOffset(player.getPos().toList(), getLookingDirection(player));
+    if (world.getBlockAt(checkPos.get(0), checkPos.get(1), checkPos.get(2)).getId() == 'lectern'){
+        found = true;
+        bx = Math.floor(checkPos.get(0));
+        by = Math.floor(checkPos.get(1));
+        bz = Math.floor(checkPos.get(2));
+    }
+
+    foreach (Sx : SxL) {
+        foreach (Sy : SyL ) {
+            foreach (Sz : SzL) {
+                if (found){
+                    break;
+                }
+                if (world.getBlockAt(Sx,Sy,Sz).getId() == 'lectern'){
+                    bx = Sx;
+                    by = Sy;
+                    bz = Sz;
+                    found = true;
+                    break;
+                }
+            }
+        }
+    }
+    if(!found) {
+        player.message("Cannot find lectern");
+    }
+    else {
+        if(localBox != null){
+            localBox.stopRendering();
+        }
+        player.message('Lectern at ' + bx + ' '+ by + ' ' + bz);
+        localBox = new BoxShape(bx,by,bz);
+        localBox.setColour(18540);
+        localBox.setOutlineColor(0,0,0);
+        localBox.setOpacity(20);
+        localBox.setOutlinePixelWidth(15);
+        localBox.setRenderThroughBlocks(true);
+        localBox.render();
+        sleep(125);
+    }
+};
+refreshLecternTask();
+
+refMap = {
+    "fortune" : {'lvl':3, 'price':14},
+    "mending" :{'lvl':1, 'price':12},
+    "unbreaking" : {'lvl':3, 'price':17},
+    "aqua_affinity" : {'lvl':1, 'price':7},
+    "blast_protection" : {'lvl':4, 'price':20},
+    "depth_strider" : {'lvl':3, 'price':12},
+    "feather_falling" : {'lvl':4, 'price':17},
+    "frost_walker" : {'lvl':1, 'price':20},
+    "protection" : {'lvl':4, 'price':20},
+    "respiration" : {'lvl':3, 'price':15},
+    "thorns" : {'lvl':3, 'price':15},
+    "efficiency" : {'lvl':5, 'price':26},
+    "sharpness" : {'lvl':5, 'price':20},
+    "smite" : {'lvl':5, 'price':25},
+    "sweeping_edge" : {'lvl':3, 'price':15},
+    "channeling" : {'lvl':1, 'price':20},
+    "flame" : {'lvl':1, 'price':10},
+    "looting" : {'lvl':3, 'price':12},
+    "infinity" : {'lvl':1, 'price':5},
+    "loyalty" : {'lvl':3, 'price':20},
+    "riptide" : {'lvl':3, 'price':20},
+    "multishot" : {'lvl':1, 'price':20},
+    "piercing" : {'lvl':4, 'price':25},
+    "power" : {'lvl':5, 'price':17},
+    "punch" : {'lvl':2, 'price':12},
+    "quick_charge" : {'lvl':3, 'price':17},
+    "luck_of_the_sea" : {'lvl':3, 'price':17},
+    "lure" : {'lvl':3, 'price':17},
+    "silk_touch" : {'lvl':1, 'price':5}
+};
+auto = null;
+mapFor = refMap;
+fun togglePaperMode(){
+    papermode = !papermode;
+    player.message('Papermode is set to ' + papermode);
+}
+fun swapToaxe() {
+    slot = player.getSlotFor(ItemStack.of(Material.of("netherite_axe")));

```

Comment:
There's lots wrong with this function but it's already in InventoryUtils so you should use that instead, but obviously fix that one first lol